modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts

```diff
@@ -0,0 +1,264 @@
+
+        console.log('initStakedReliquaryBalances: performing db operations...');
+
+        await prismaBulkExecuteOperations(
+            [
+                prisma.prismaUser.createMany({
+                    data: userAddresses.map((userAddress) => ({ address: userAddress })),
+                    skipDuplicates: true,
+                }),
+                prisma.prismaUserStakedBalance.deleteMany({}),
+                prisma.prismaUserStakedBalance.createMany({
+                    data: relics.map((relic) => {
+                        const pool = pools.find((pool) => isSameAddress(pool.address, relic.pool.poolTokenAddress));
+
+                        return {
+                            id: relic.id,
+                            balance: formatFixed(relic.balance, 18),
+                            balanceNum: parseFloat(formatFixed(relic.balance, 18)),
+                            userAddress: relic.userAddress,
+                            poolId: pool?.id,
+                            tokenAddress: relic.pool.poolTokenAddress,
+                            stakingId: `${relic.pid}`,
+                        };
+                    }),
+                }),
+                prisma.prismaUserBalanceSyncStatus.upsert({
+                    where: { type: 'RELIQUARY' },
+                    create: { type: 'RELIQUARY', blockNumber: block.number },
+                    update: { blockNumber: block.number },
+                }),
+            ],
+            true,
+        );
+
+        console.log('initStakedReliquaryBalances: finished...');
+    }
+
+    public async syncUserBalance({ userAddress, poolId, poolAddress, staking }: UserSyncUserBalanceInput) {
+        if (staking.type !== 'RELIQUARY') {
+            return;
+        }
+        const reliquary: Reliquary = getContractAt(networkConfig.reliquary.address, ReliquaryAbi);
+        const relicPositions = await reliquary.relicPositionsOfOwner(userAddress);
+
+        const relicIds = relicPositions[0];
+        const positions = relicPositions[1];
+        for (let i = 0; i < relicIds.length; i++) {
+            if (positions[i].poolId.toString() !== poolId) {
+                continue;
+            }
+
+            const amountStaked = formatFixed(positions[i].amount, 18);
+            await prisma.prismaUserStakedBalance.upsert({
+                where: { id: `reliquary-${staking.id}-${userAddress}` },
+                update: {
+                    balance: amountStaked,
+                    balanceNum: parseFloat(amountStaked),
+                },
+                create: {
+                    id: `reliquary-${staking.id}-${userAddress}`,
+                    balance: amountStaked,
+                    balanceNum: parseFloat(amountStaked),
+                    userAddress,
+                    poolId: poolId,
+                    tokenAddress: poolAddress,
+                    stakingId: staking.id,
+                },
+            });
+        }
```

Comment:
Doesn't work if a user has more than one relic for the same pool. Also need to add all amounts.

Reply:
yea mixed some thoughts here, this should be better i think :)